cmd/snap/cmd_buy.go

```diff
@@ -117,33 +110,28 @@ Once completed, return here and run 'snap buy %s' again.`), snap.Name)
 		return err
 	}
 
-	reader := bufio.NewReader(nil)
-	reader.Reset(Stdin)
+	fmt.Fprintf(Stdout, i18n.G(`Please re-enter your Ubuntu One password to purchase '%s' from '%s' for %s. Press ctrl-c to cancel.`),
```

Comment:
That string is too long to be reasonable to paste to the terminal as is; cut it into lines, below 80 wide, please. You can use `ToText` (from `"go/doc"`) if you don't want to do it by hand.

Also, use `%q` not `'%s'`.